ext/mixed/js/core.js

```diff
@@ -269,17 +269,26 @@ const yomichan = (() => {
         constructor() {
             super();
 
+            this._isBackendPreparedResolve = null;
+            this._isBackendPreparedPromise = new Promise((resolve) => (this._isBackendPreparedResolve = resolve));
+
             this._messageHandlers = new Map([
+                ['backendPrepared', this._onBackendPrepared.bind(this)],
                 ['getUrl', this._onMessageGetUrl.bind(this)],
                 ['optionsUpdated', this._onMessageOptionsUpdated.bind(this)],
                 ['zoomChanged', this._onMessageZoomChanged.bind(this)]
             ]);
 
             chrome.runtime.onMessage.addListener(this._onMessage.bind(this));
+            chrome.runtime.sendMessage({action: 'yomichanOnline'});
```

Comment:
core.js is used in the background window also, and you won't want this called from there. Also, a more descriptive name might be better, like `'yomichanCoreReady'` or something.

Reply:
This should be fixed in https://github.com/FooSoft/yomichan/pull/386/commits/56b2f2c853494b228aa4467a64ed91486432bffd.

The reason why `'backendPrepared'` can be triggered from two places is that `Backend` might not be up when `'yomichanCoreReady'` is sent. I'm not sure if it's possible that both events are missed by `Yomichan`, but I haven't seen that happen yet. Logically no, but browsers aren't always perfect.